harness/determined/pytorch/_pytorch_context.py

```diff
@@ -51,9 +51,18 @@ class PyTorchTrialContext(det.TrialContext, pytorch._PyTorchReducerContext):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         det.TrialContext.__init__(self, *args, **kwargs)
         pytorch._PyTorchReducerContext.__init__(self, self.distributed.allgather)
+
+        self.cluster_info = det.get_cluster_info()
+        if self.cluster_info:
+            self.managed_training = True
+        else:
+            self.managed_training = False
+
+        self.experiment_config = self.cluster_info.trial._config
```

Comment:
Having a `self.experiment_config` on the PytorchTrialContext is sort of antithetical to the goal of supporting local training mode.

How hard/impossible is it to remove it entirely?  What would it take to get there?